DT-6663 update transfer time bubbles on map

vesameskanen opened the pull request:
Navigation view uses a callback function to update the itinerary page's stored itinerary.  When itinerary changes, itinerary map will re-render, showing possibly changed transfer times.

vesameskanen committed:
fix: update transfer time bubbles on map in navi view

vesameskanen committed:
fix: remove dead props

vesameskanen committed:
fix: route number centering in navi transfers

partisaani commented on app/component/itinerary/ItineraryPage.js:
```diff
@@ -696,6 +696,16 @@ export default function ItineraryPage(props, context) {
     setStoredItinerary(itineraryWithParams);
   };
 
+  const updateStoredItinerary = legs => {
+    setStoredItinerary({
+      ...storedItinerary,
+      itinerary: {
+        ...storedItinerary.itinerary,
+        legs,
+      },
+    });
+  };
+
```
Should this also update the itinerary we've stored in local storage?

vesameskanen replied:
I don't think so. Continuous writing to localStorage feels bad and should be avoided unless there is a good reason.

partisaani commented on app/component/itinerary/navigator/hooks/useRealtimeLegs.js:
```diff
@@ -220,7 +225,7 @@ const useRealtimeLegs = (relayEnvironment, initialLegs, position) => {
         l.freezeEnd = l.freezeEnd || legTime(l.end) <= now;
       });
-
+      updateLegs(rtLegs);
```
While not probable, it is possible for `updateLegs` to be undefined. Should be surrounded with if

vesameskanen replied:
UpdateLegs is defined as a required prop. How can it be undefined? If it is, code should crash, right?

partisaani replied:
That's true, but such safeguard is missing from the hook's point of view.

vesameskanen committed:
chore: validate leg update callback

vesameskanen committed:
feat: render navigator transfer times green

For some reason, className passed to leaflet does not work.
Transfer time is changed green using inline style.
